src/Sylius/Component/Core/spec/Promotion/Checker/Rule/ContainsProductRuleCheckerSpec.php

```diff
@@ -32,89 +37,44 @@ function it_is_initializable()
         $this->shouldHaveType(ContainsProductRuleChecker::class);
     }
 
-    function it_is_a_rule_checker()
+    function it_implements_a_rule_checker_interface()
     {
         $this->shouldImplement(RuleCheckerInterface::class);
     }
 
-    function it_throws_an_exception_on_invalid_subject(PromotionSubjectInterface $subject)
+    function it_throws_an_exception_if_the_promotion_subject_is_not_an_order(PromotionSubjectInterface $subject)
     {
-        $this->shouldThrow(UnsupportedTypeException::class)->duringIsEligible($subject, []);
+        $this
+            ->shouldThrow(new UnexpectedTypeException($subject->getWrappedObject(), OrderInterface::class))
+            ->during('isEligible', [$subject, []])
+        ;
     }
 
-    function it_returns_true_if_variant_is_right_and_exclude_is_not_set(
+    function it_returns_true_if_product_is_right(
```

Comment:
It would be nice to have two products here and the wanted one as the second so we will be sure that we are checking every item.